### PR TITLE
Add customFields for 3DS verifyCard

### PR DESCRIPTION
**Pull request by warmkesselj**
### Summary of changes

Accept a raw JSON format for customFields in BTThreeDSecureRequest for VerifyCard.

### Checklist

- [X] Added a changelog entry

### Authors
@agedd 
@jwarmkessel 
@warmkesselj

### CODE REVIEW

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift [hunk 1]
```diff
@@ -11,6 +11,9 @@ import BraintreeCore
 
     /// A nonce to be verified by ThreeDSecure
     public var nonce: String?
+    
+    /// Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+    public var customFields: [String: Any]? = [:]
```
Comment (scannillo) [inline Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift:16]: Similar comment [to here](https://github.com/braintree/braintree_android/pull/1044/files#r1650990591), with making this a more standard type that merchants use to model JSON - `Encodable`. A dictionary will also work though.

Which do you think is better merchant dev-ex?

File: Demo/Application/Features/ThreeDSecureViewController.swift [hunk 2]
```diff
@@ -91,6 +91,8 @@ class ThreeDSecureViewController: PaymentButtonBaseViewController {
 
     private func createThreeDSecureRequest(with nonce: String) -> BTThreeDSecureRequest {
         let request = BTThreeDSecureRequest()
+        
+        request.customFields = ["poop": "poop"]
```
Comment (sshropshire) [inline Demo/Application/Features/ThreeDSecureViewController.swift:95]:
```suggestion
        request.customFields = ["💩": "💩"]
```

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift (hunk 1 above)
Comment (warmkesselj) [reply Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift:16]: I just got more information that we'll be supporting no nested JSON values and n number of custom values. I'm waiting on clarification to determine if we're expecting only String values.

Comment (warmkesselj) [reply Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift:16]: Confirmed we can enforce string key values. I think [String: String] is fine. The merchant will be setting these values in their control panel. Btw, the Cardinal team is taking anything but stringifying everything coming through the gateway.

**Review by sshropshire — APPROVED** [review]

**Review by KunJeongPark — APPROVED** [review]
Great job Justin!

**Pull request merged**